affinity/cpp-20/d0796r3.md

```diff
@@ -629,16 +684,19 @@ The free function `this_thread::get_resource` is provided for retrieving the `ex
 
 *Returns:* The `execution_resource` underlying the current thread of execution.
 
+> [*Note:* The `execution_resource` underlying the current thread of execution may not necessarily be reachable from "top-level" resources visible through `this_system`. *--end note*]
```

Comment:
Is not clear to me what is the reason for the note (i have been on holiday so apologies if this has been discussed already)

Reply:
One of the things we discussed was whether some resources might not be accessible from the top level -- e.g., some GPU resources (like scratch memory) might only be available during execution on the GPU.  One could thus access them through `this_thread::get_resource` (and possibly in other ways), but not from the top level `this_system::get_resource`.

This is also one of the reasons for `execution_resource` not to be reference counted (at least not all `execution_resource`s).  Otherwise, it would be bad for them to escape the scope of code executing on the GPU.

I was thinking about this Note last night, and realized that we may need more explanatory text about this.  I'd like the PR to go through so it doesn't step on other PRs, but we should definitely write more about this.